examples/with-firebase-authentication/utils/auth/useUser.js

```diff
@@ -27,12 +30,26 @@ const useUser = () => {
   }
 
   useEffect(() => {
-    const cookie = cookies.get('auth')
-    if (!cookie) {
+    // Firebase updates the id token every hour, this
+    // makes sure the react state and the cookie are
+    // both kept up to date
+    firebase.auth().onIdTokenChanged((user) => {
+      if (user) {
+        const userData = mapUserData(user)
+        setUserCookie(userData)
+        setUser(userData)
+      } else {
+        removeUserCookie()
+        setUser()
+      }
+    })
+
+    const userFromCookie = getUserFromCookie()
```

Comment:
This is the user retrieval from the cookie that existed before. It's not really necessary now, as it will be handled by `onIdTokenChanged` above, but I've left it in because I think it's a bit quicker to load than the firebase client library is 🤷‍♂️